pkg/upgrade/upgrade_test.go

```diff
@@ -134,10 +135,38 @@ func TestSkipForNonOwnedInstances(t *testing.T) {
 	cl := fake.NewFakeClient(objs...)
 
 	// test
-	assert.NoError(t, ManagedInstances(context.Background(), cl, cl))
+	assert.NoError(t, ManagedInstances(context.Background(), cl, cl, opver.Get().Jaeger))
 
 	// verify
 	persisted := &v1.Jaeger{}
 	assert.NoError(t, cl.Get(context.Background(), nsn, persisted))
 	assert.Equal(t, "1.11.0", persisted.Status.Version)
 }
+
+func TestSkipForInvalidSemVer(t *testing.T) {
+	invalidVersion := "xxx...xx"
+
+	// prepare
+	nsn := types.NamespacedName{Name: "my-instance"}
+
+	existing := v1.NewJaeger(nsn)
+	existing.Status.Version = "1.11.0" // this is the first version we have an upgrade function
+	objs := []runtime.Object{existing}
+
+	s := scheme.Scheme
+	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Jaeger{})
+	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.JaegerList{})
+	cl := fake.NewFakeClient(objs...)
+	versions[invalidVersion] = version{v: invalidVersion, upgrade: noop}
+	defer func() {
+		delete(versions, invalidVersion)
+	}()
+	// test
+	assert.NoError(t, ManagedInstances(context.Background(), cl, cl, opver.Get().Jaeger))
```

Comment:
Shouldn't this return an error? Due to parsing invalid version.

Reply:
Mmm not sure about this, the original behavior of `ManagedInstances` is to skip the jaeger instance in case of errors. and due the fact that I'm parsing the versions on `ManagedInstance`, it will fail without returning error here.

May be I should move that logic to other place.

Reply:
There is another way, I can assume that the array is sorted (we need to always make sure that is the case when we added upgrade routines) , and create the semver directly instead of parsing. WDYT?

Reply:
Ok, no, I cannot find an official way to do that with the library I'm using, only the parser function. of course I can create the struct and fill the fields, but not sure that will be appropriate.